woocommerce/woocommerce
# Suggest product categories with AI

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds the functionality to suggest product categories using AI. Currently, it can generate suggestions based on the available categories only. Meaning that it will check the list of available categories in the store and suggest which ones are a good match for the product.

Once suggestions are returned, a list is displayed to the user where they can select to add each suggestion.

The reason why it's only suggesting existing categories instead of allowing the user to create new categories is because we do not have an easy method to create multi-level categories (Parent > Child) on the product edit page through JS. This can be revisited in another PR, but since the amount of work to implement it was substantial, I skipped it for the MVP phase.

As part of #39436



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new product or edit an existing one
2. Make sure you have some categories defined in your store
3. Enter some data (title, description, tags, etc.) for your product
4. Click "Suggest categories using AI" on the category meta box
5. Confirm that the categories returned are a good match for the product based on your existing categories
6. Select some categories to add them to your product
7. Update the product and confirm the categories will be added to it
8. Edit the product again and click on the suggest categories button
9. This time, it should not return any categories that you previously selected (or are already assigned to the product). It should instead display a message indicating that your product is well-categorized.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
